From ae2735b166a37dbfae0774e96e4bf91ce9de4c4d Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Sat, 17 Apr 2021 10:08:28 +0000
Subject: [PATCH] libstdc++: fix ostream operator<< with std::unique_ptr

std::unique_ptr has no operator<< in libstdc++. Even it would
exist, it would just return the pointer value and not the content.
Use operator* instead, but also check that the pointer is not empty.
---
 .../enterprise/connectors/file_system/box_api_call_flow.cc  | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/chrome/browser/enterprise/connectors/file_system/box_api_call_flow.cc b/chrome/browser/enterprise/connectors/file_system/box_api_call_flow.cc
index e9544d5..00c6b3c 100644
--- a/chrome/browser/enterprise/connectors/file_system/box_api_call_flow.cc
+++ b/chrome/browser/enterprise/connectors/file_system/box_api_call_flow.cc
@@ -325,8 +325,10 @@ void BoxPreflightCheckApiCallFlow::ProcessApiCallFailure(
     std::unique_ptr<std::string> body) {
   auto response_code = head->headers->response_code();
   DLOG(ERROR) << "[BoxApiCallFlow] PreflightCheck failed; net error = "
-              << net_error << "; response code = " << response_code << ": "
-              << body;
+              << net_error << "; response code = " << response_code;
+  if (!body->empty()) {
+    DLOG(ERROR) << "Body: " << *body;
+  }
   std::move(callback_).Run(false, response_code);
 }
 
-- 
2.26.3

